File: parse_pois.py
import re
import json

def extract_pois(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    pois = []
    
    # We will find all occurrences of `id: "..."`
    for match in re.finditer(r'id:\s*["\']([^"\']+)["\']', content):
        poi_id = match.group(1)
        # find the name
        name_match = re.search(r'name:\s*\{[^\}]*en:\s*["\']([^"\']+)["\']', content[match.start():match.start()+1000])
        name = name_match.group(1) if name_match else poi_id
        pois.append({'id': poi_id, 'name': name})
    return pois

pois1 = extract_pois('lib/visualLab/data/polandPoi.ts')
pois2 = extract_pois('lib/visualLab/data/poiExtraPolandCities.ts')

with open('pois_to_do.json', 'w', encoding='utf-8') as f:
    json.dump({'polandPoi': pois1, 'poiExtraPolandCities': pois2}, f, indent=2)

print(f"Found {len(pois1)} in polandPoi.ts and {len(pois2)} in poiExtraPolandCities.ts")

File: parse_pois.js
const fs = require('fs');

const files = [
  "lib/visualLab/data/poiExtraNigeriaCitiesV2.ts",
  "lib/visualLab/data/poiExtraNigeriaEconomicV2.ts",
  "lib/visualLab/data/poiExtraNigeriaHistoryV2.ts",
  "lib/visualLab/data/poiExtraNigeriaLandmarksV2.ts",
  "lib/visualLab/data/poiExtraNigeriaLifeV2.ts",
  "lib/visualLab/data/poiExtraNigeriaNatureV2.ts",
  "lib/visualLab/data/poiExtraNigeriaReliefV2.ts"
];

const missingPois = [];

files.forEach(file => {
  const content = fs.readFileSync(file, 'utf-8');
  const regex = /id:\s*"([^"]+)"([\s\S]*?)(?=id:\s*"|\];)/g;
  let match;
  while ((match = regex.exec(content)) !== null) {
    const id = match[1];
    const block = match[2];
    
    let hasDescAdvEn = false;
    const descAdvMatch = block.match(/descriptionAdvanced:\s*\{([\s\S]*?)\}/);
    if (descAdvMatch) {
      const enMatch = descAdvMatch[1].match(/en:\s*"(.*?)"/);
      if (enMatch && enMatch[1].trim() !== '') {
        hasDescAdvEn = true;
      }
    }
    
    if (!hasDescAdvEn) {
      // get name.en
      const nameMatch = block.match(/name:\s*\{[^}]*en:\s*"([^"]+)"/);
      const nameEn = nameMatch ? nameMatch[1] : id;
      // get description.en
      const descMatch = block.match(/description:\s*\{[^}]*en:\s*"([^"]+)"/);
      const descEn = descMatch ? descMatch[1] : "";
      
      missingPois.push({ id, name: nameEn, desc: descEn });
    }
  }
});

fs.writeFileSync('missing_pois.json', JSON.stringify(missingPois, null, 2));
console.log(`Found ${missingPois.length} missing POIs.`);
